src/3d/qgspointcloudlayer3drenderer.h

```diff
@@ -170,6 +171,20 @@ class _3D_NO_EXPORT QgsPointCloud3DRenderContext : public Qgs3DRenderContext
      */
     QgsCoordinateTransform coordinateTransform() const { return mCoordinateTransform; }
 
+    /**
+     * Cancels rendering by emitting renderingCanceled signal
+     * \since QGIS 3.20
+     */
+    void cancelRendering();
+
+  signals:
+
+    /**
+     * Emitted when the rendering is canceled
+     * \since QGIS 3.20
+     */
+    void renderingCanceled();
```

Comment:
Rather than making Qgs3DRenderContext a QObject and adding a signal about canceled rendering, I would prefer to add QgsFeedback object to QgsPointCloud3DRenderContext, which is QObject-based and provides interface for cancellation (including a canceled() signal). So the client code would just call `mContext.feedback()->cancel()`